### PR TITLE
RAITO-2454 | Import database roles from Snowflake with what + who locked

### PR DESCRIPTION
**Pull request by TimmersThomas**

### CODE REVIEW

File: snowflake/constants.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ const (
 	SfSkipTags                          = "sf-skip-tags"
 	SfSkipColumns                       = "sf-skip-columns"
 	SfDataUsageWindow                   = "sf-data-usage-window"
+	SfDatabaseRoles                     = "sf-database-roles"
```
Comment (codatoz) [inline snowflake/constants.go:17]: Needs to be added with its description to `main.go`

File: snowflake/data_access_sync_from_target.go [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+
+	ap, f := processedAps[externalId]
+	if !f {
+		processedAps[externalId] = &exporter.AccessProvider{
+			Type:       currentApType,
+			ExternalId: externalId,
+			// Updated this because of https://github.com/raito-io/appserver/blob/587484940a2e356a486dd8779166852761885353/lambda/appserver/services/access_provider/importer/importer.go#L523
+			ActualName: roleName,
+
+			Name:       roleName,
```
Comment (codatoz) [inline snowflake/data_access_sync_from_target.go:204]: Wondering if we should add the data source name to the (display) name of the AP to make it more distinguishable in the UI. Thoughts?

**Review by codatoz — CHANGES_REQUESTED** [review]
Small remark

File: snowflake/data_access_sync_from_target.go (hunk 2 above)
Comment (TimmersThomas) [reply snowflake/data_access_sync_from_target.go:204]: Was indeed also one of my toughts, when using it currently in the UI you don't have any clue within what scope this is.

File: snowflake/data_access_sync_to_target.go [hunk 3]
```diff
@@ -0,0 +1,1659 @@
...
+	return schemas, nil
+}
+
+func (s *AccessSyncer) getDatabases(repo dataAccessRepository) ([]DbEntity, error) {
+	if s.databasesCache != nil {
+		return s.databasesCache, nil
+	}
+
+	var err error
+	s.databasesCache, err = repo.GetDatabases()
```
Comment (TimmersThomas) [inline snowflake/data_access_sync_to_target.go:1156]: Just wondering, should we take exclude-database items into account?

If not, should be do an extra filtering for the databases we are going to loop over for getting the database-roles based on that parameter?

**Review by rmennes — DISMISSED** [review]

File: snowflake/data_access_sync_to_target.go (hunk 3 above)
Comment (codatoz) [reply snowflake/data_access_sync_to_target.go:1156]: We should definitely only look at the database we are importing, yes. So just like for the data-source part, we need to filter out the right databases here.

**Pull request merged**